pkg/cli/cli.go

```diff
@@ -0,0 +1,187 @@
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	apiv1 "github.com/google/cloud-android-orchestration/api/v1"
+
+	"github.com/spf13/cobra"
+)
+
+// Groups streams for standard IO.
+type IOStreams struct {
+	In     io.Reader
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type CommandOptions struct {
+	IOStreams
+	Args []string
+}
+
+type CVDRemoteCommand struct {
+	command *cobra.Command
+}
+
+func NewCVDRemoteCommand() *CVDRemoteCommand {
+	opts := &CommandOptions{
+		IOStreams: IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr},
+		Args:      os.Args[1:],
+	}
+	return NewCVDRemoteCommandWithArgs(opts)
+}
+
+const (
+	serviceURLFlag = "service_url"
+)
+
+type configFlags struct {
+	ServiceURL string
+}
+
+func NewCVDRemoteCommandWithArgs(o *CommandOptions) *CVDRemoteCommand {
+	var configFlags = new(configFlags)
+	hostCmd := &cobra.Command{
+		Use:   "host",
+		Short: "Work with hosts",
+	}
+	hostCmd.AddCommand(newCreateHostCommand())
+	rootCmd := &cobra.Command{
+		Use:               "cvdremote",
+		Short:             "Manages Cuttlefish Virtual Devices (CVDs) in the cloud.",
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+		CompletionOptions: cobra.CompletionOptions{DisableDefaultCmd: true},
+	}
+	rootCmd.SetArgs(o.Args)
+	rootCmd.SetOut(o.IOStreams.Out)
+	rootCmd.SetErr(o.IOStreams.ErrOut)
+	rootCmd.PersistentFlags().StringVar(&configFlags.ServiceURL, serviceURLFlag, "",
+		"Cloud orchestration service url.")
+	rootCmd.MarkPersistentFlagRequired(serviceURLFlag)
+	// Do not show a `help` command, users have always the `-h` and `--help` flags for help purpose.
+	rootCmd.SetHelpCommand(&cobra.Command{Hidden: true})
+	rootCmd.AddCommand(hostCmd)
+	return &CVDRemoteCommand{rootCmd}
+}
+
+// ExecuteNoErrOutput is a version of Execute which returns the command error instead of
+// printing it and do not exit in case of error.
+func (c *CVDRemoteCommand) ExecuteNoErrOutput() error {
+	return c.command.Execute()
+}
+
+func (c *CVDRemoteCommand) Execute() {
+	if err := c.ExecuteNoErrOutput(); err != nil {
+		c.command.PrintErrln(err)
+		os.Exit(1)
+	}
+}
+
+type opTimeoutError string
+
+func (s opTimeoutError) Error() string {
+	return fmt.Sprintf("waiting for operation %q timed out", string(s))
+}
+
+type apiCallError struct {
+	Err *apiv1.Error
+}
+
+func (e *apiCallError) Error() string {
+	return fmt.Sprintf("api call error %s: %s", e.Err.Code, e.Err.Message)
+}
+
+func newCreateHostCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates a host.",
+		RunE: func(c *cobra.Command, args []string) error {
+			return runCreateHostCommand(c)
+		},
+	}
+	return cmd
+}
+
+func runCreateHostCommand(c *cobra.Command) error {
+	serviceURL := c.InheritedFlags().Lookup(serviceURLFlag).Value.String()
+	baseURL := serviceURL + "/v1"
+	req := &apiv1.CreateHostInstanceRequest{}
+	body := &apiv1.CreateHostRequest{CreateHostInstanceRequest: req}
+	op := new(apiv1.Operation)
```

Comment:
ni: same comment as below about `op` not having to be a pointer.

Reply:
done